supervision/detection/line_counter.py

```diff
@@ -26,16 +26,61 @@ class LineZone:
             to outside.
     """
 
-    def __init__(self, start: Point, end: Point):
+    def __init__(self, start: Point, end: Point, count_condition="whole_crossed"):
         """
         Args:
             start (Point): The starting point of the line.
             end (Point): The ending point of the line.
+            count_condition (str): The condition which determines
+                how detections are counted as having crossed the line
+                counter. Can either be "whole_crossed" or "center_point_crossed".
+
+                If condition is set to "whole_crossed", trigger() determines
+                whether if the whole bounding box of the detection has crossed
+                the line or not. This is the default behaviour.
+
+                If condition is set to "center_point_crossed", trigger() determines
+                whether if the center point of the detection's bounding box has
+                crossed the line or not.
         """
         self.vector = Vector(start=start, end=end)
         self.tracker_state: Dict[str, bool] = {}
         self.in_count: int = 0
         self.out_count: int = 0
+        self.count_condition = count_condition
+        if count_condition not in ["whole_crossed", "center_point_crossed"]:
+            raise ValueError(
+                "Argument count_condition must be 'whole_crossed' or 'center_point_crossed'"
+            )
+
+    def is_point_in_line_range(self, point: Point) -> bool:
```

Comment:
The given function works for the above case of a diagonal line. In the case of purely vertical and horizontal lines; Xstart, Xend and Ystart, Yend respectively are the same. So in those cases only the coordinate range that is meaningful to check is being asserted. In all other cases (such as the diagonal line in the image) both coordinate ranges should hold for the point of interest to be considered "in range".

Reply:
However, it should be noted that in the current way the function works, the dashed lines are parallel to the axes instead of being perpendicular to the line as shown above. So in reality the area considered as "in range" of the line counter's coordinates is much smaller. This could potentially lead to some bugs when there are multiple triggering anchors that are far from each other, in which case the line counter won't count the object as having crossed in or out.

Reply:
Thinking of a fix for that in the mean time :)